aten/src/ATen/templates/Functions.cpp

```diff
@@ -7,41 +7,47 @@
 namespace at {
 
 Tensor TensorMaker::make_tensor() {
-   AutoDispatchBelowADInplaceOrView guard{}; // TODO: Remove.
-   tracer::impl::NoTracerDispatchMode tracer_guard{};
+  AutoDispatchBelowADInplaceOrView guard{}; // TODO: Remove.
+  tracer::impl::NoTracerDispatchMode tracer_guard{};
 
-   check_size_nonnegative(sizes_);
+  check_size_nonnegative(sizes_);
 
-   TORCH_CHECK_VALUE(
-       !deleter_ || !ctx_,
-       "The deleter and context arguments are mutually exclusive.");
+  TORCH_CHECK_VALUE(
+      !deleter_ || !ctx_,
+      "The deleter and context arguments are mutually exclusive.");
 
-   if (device_ == nullopt) {
-     device_ = globalContext().getDeviceFromPtr(data_, opts_.device().type());
-   }
+  if (device_ == nullopt) {
+    device_ = globalContext().getDeviceFromPtr(data_, opts_.device().type());
+  }
 
-   if (opts_.device().has_index()) {
-     // clang-format off
-     TORCH_CHECK_VALUE(
-         opts_.device() == *device_,
-         "Specified device ", opts_.device(), " does not match device of data ", *device_);
-     // clang-format on
-   }
+  if (opts_.device().has_index()) {
+    // clang-format off
+    TORCH_CHECK_VALUE(
+        opts_.device() == *device_,
+        "Specified device ", opts_.device(), " does not match device of data ", *device_);
+    // clang-format on
+  }
 
-   std::size_t size_bytes = computeStorageSize();
+  std::size_t size_bytes = computeStorageSize();
 
-   DataPtr data_ptr{};
-   if (deleter_) {
-     data_ptr = makeDataPtrFromDeleter();
-   } else {
-     data_ptr = makeDataPtrFromContext();
-   }
+  if (data_ == nullptr) {
+    // We need to ensure that there's always a valid pointer or the underlying
+    // std::unique_ptr<> won't call the deleter (custom or context).
+    data_ = malloc(0);
```

Comment:
Won't this leak if not matched by a corresponding `free()` call? How about setting this to some dummy value like `(void *) 0x1` instead?

Reply:
It's not a dummy value - it gets turned into a std::unique_ptr -- but tracing through the code again I'm not sure why (in the deleter case) it isn't actually freed twice since it's passed to two different std::unique_ptrs.  I'm going to convert this to a draft until I convince myself what is actually going on.

Reply:
Ok - I see my misunderstanding. `UniqueVoidPtr(p)` doesn't actually delete `p` when it goes out of scope. You're right - this value is essentially a dummy which won't be deleted.

Reply:
Updated to not use the dummy allocation - this is a cleaner solution.